lib/std/math/prob/gamma.zig

```diff
@@ -0,0 +1,368 @@
+// SPDX-License-Identifier: MIT
+// Copyright (c) 2015-2021 Zig Contributors
+// This file is part of [zig](https://ziglang.org/), which is MIT licensed.
+// The MIT license requires this copyright notice to be included in all copies
+// and substantial portions of the software.
+//
+// Ported from the Cephes library. Original license below:
+//
+// Cephes Math Library Release 2.8:  June, 2000
+// Copyright 1984, 1987, 1989, 1992, 2000 by Stephen L. Moshier
+
+const std = @import("../../std.zig");
+const math = std.math;
+
+usingnamespace @import("constants.zig");
+
+const polevl = math.prob.polevl;
+const p1evl = math.prob.p1evl;
+
+const P = [_]f64{
+    1.60119522476751861407E-4,
+    1.19135147006586384913E-3,
+    1.04213797561761569935E-2,
+    4.76367800457137231464E-2,
+    2.07448227648435975150E-1,
+    4.94214826801497100753E-1,
+    9.99999999999999996796E-1,
+};
+
+const Q = [_]f64{
+    -2.31581873324120129819E-5,
+    5.39605580493303397842E-4,
+    -4.45641913851797240494E-3,
+    1.18139785222060435552E-2,
+    3.58236398605498653373E-2,
+    -2.34591795718243348568E-1,
+    7.14304917030273074085E-2,
+    1.00000000000000000320E0,
+};
+
+const MAXGAM = 171.624376956302725;
+const LOGPI = 1.14472988584940017414;
+
+// Stirling's formula for the gamma function
+const STIR = [_]f64{
+    7.87311395793093628397E-4,
+    -2.29549961613378126380E-4,
+    -2.68132617805781232825E-3,
+    3.47222221605458667310E-3,
+    8.33333333333482257126E-2,
+};
+
+const MAXSTIR = 143.01608;
+const SQTPI = 2.50662827463100050242E0;
+
+/// Gamma function computed by Stirling's formula.
+/// The polynomial STIR is valid for 33 <= x <= 172.
+fn stirf(x: f64) f64 {
+    var w = 1.0 / x;
+    w = 1.0 + w * polevl(w, STIR[0..]);
+    var y = math.exp(x);
+
+    if (x > MAXSTIR) {
+        // Avoid overflow in pow()
+        var v = math.pow(f64, x, 0.5 * x - 0.25);
+        y = v * (v / y);
+    } else {
+        y = math.pow(f64, x, x - 0.5) / y;
+    }
+
+    y = SQTPI * y * w;
+    return y;
+}
+
+fn small(x: f64, z: f64) f64 {
+    if (x == 0.0) {
+        return math.nan(f64); // Domain
+    }
+    return z / ((1.0 + 0.5772156649015329 * x) * x);
+}
+
+/// Gamma function
+///
+/// Returns gamma function of the argument.  The result is
+/// correctly signed, and the sign (+1 or -1) is also
+/// returned in a global (extern) variable named sgngam.
```

Comment:
I don't see a global extern variable here.... but also that sounds like a terrible idea

Reply:
Yeah I'll need to update the documentation. It is taken verbatim from the C source mostly.